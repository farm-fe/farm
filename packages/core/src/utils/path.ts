import path from 'node:path';

const windowsSlashRE = /\\/g;
export function slash(p: string): string {
  return p.replace(windowsSlashRE, '/');
}

export function withTrailingSlash(path: string): string {
  if (path[path.length - 1] !== '/') {
    return `${path}/`;
  }
  return path;
}

const postfixRE = /[?#].*$/;
export function stripQueryAndHash(path: string): string {
  return path.replace(postfixRE, '');
}

export function removeHashFromPath(path: string): string {
  const hashRE = /([_-][a-f0-9]{4,12})(\.[^./]+(\.[^./]+)*)$/;
  return path.replace(hashRE, '$2');
}

export function getShortName(file: string, root: string): string {
  const result = file.startsWith(withTrailingSlash(root))
    ? path.relative(root, file)
    : file;
  return result;
}
