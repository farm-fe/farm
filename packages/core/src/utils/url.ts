const postfixRE = /[?#].*$/;
export function cleanUrl(url: string): string {
  return url.replace(postfixRE, '');
}

const importQueryRE = /(\?|&)import=?(?:&|$)/;
export const isImportRequest = (url: string): boolean =>
  importQueryRE.test(url);

const trailingSeparatorRE = /[?&]$/;
export function removeImportQuery(url: string): string {
  return url.replace(importQueryRE, '$1').replace(trailingSeparatorRE, '');
}

export const knownJavascriptExtensionRE = /\.[tj]sx?$/;

export const urlRE = /(\?|&)url(?:&|$)/;
