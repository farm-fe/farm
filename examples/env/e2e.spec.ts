import { test, expect } from 'vitest';
import { startProjectAndTest } from '../../e2e/vitestSetup';
import { basename, dirname } from 'path';
import { fileURLToPath } from 'url';

const name = basename(import.meta.url);
const projectPath = dirname(fileURLToPath(import.meta.url));

test(`e2e tests - ${name}`, async () => {
  const runTest = (command?: 'start' | 'preview') =>
    startProjectAndTest(
      projectPath,
      async (page) => {
        const rootHtml = await page.innerHTML('body');
        expect(rootHtml).matchSnapshot();
      },
      command
    );

  await runTest();
  await runTest('preview');
});
