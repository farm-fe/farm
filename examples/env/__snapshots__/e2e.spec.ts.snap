// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`e2e tests - e2e.spec.ts 1`] = `
"
    <h1>Environment Variables</h1>
    <p>FARM_CUSTOM_ENV_VARIABLE: <code class="custom">1</code></p>
    <p>FARM_EFFECTIVE_MODE_FILE_NAME: <code class="file">.env.development</code></p>
    <p>FARM_BOOL: <code class="bool">true</code></p>
    <p>FARM_EXPAND_A: <code class="expand_a">8asdasd</code></p>
    <p>CUSTOM_PREFIX_ENV_VARIABLE: <code class="prefix">1</code></p>
    <p>
      FARM_EXPAND_B:
      <code class="expand_b"></code>
    </p>
    <p>
      FARM_ESCAPE_A:
      <code class="escape_a">escape$</code>
    </p>
    <p>FARM_ESCAPE_B: <code class="escape_b">escape$</code></p>
    <p>
      FARM_CUSTOM_ENV_VARIABLE(globalThis process env):
      <code class="globalThis">1</code>
    </p>
    <p>
      FARM_CUSTOM_ENV_VARIABLE(global process env): <code class="global">1</code>
    </p>
    <script>
      // console.log(process.env);
      text('.custom', "1");
      text('.file', ".env.development");
      text('.bool', "true");
      text('.prefix', "1");
      text('.expand_a', "8asdasd");
      text('.expand_b', "");
      text('.escape_b', "escape$");
      text('.escape_a', "escape$");
      text('.globalThis', "1");
      text('.global', "1");
      function text(el, text) {
        document.querySelector(el).textContent = text;
      }
    </script>
  

<script data-farm-entry-script="true">(globalThis || window || global)['4c4d99f12bb13ca32da8c12d962efd60'].__farm_module_system__.setInitialLoadedResources([]);</script><script data-farm-entry-script="true">(globalThis || window || global)['4c4d99f12bb13ca32da8c12d962efd60'].__farm_module_system__.setDynamicModuleResourcesMap({  });</script><script data-farm-entry-script="true">(globalThis || window || global)['4c4d99f12bb13ca32da8c12d962efd60'].__farm_module_system__.setPublicPaths(['/']);</script><script data-farm-entry-script="true">(globalThis || window || global)['4c4d99f12bb13ca32da8c12d962efd60'].__farm_module_system__.bootstrap();</script>"
`;

exports[`e2e tests - e2e.spec.ts 2`] = `
"<h1>Environment Variables</h1>
    <p>FARM_CUSTOM_ENV_VARIABLE: <code class="custom">2</code></p>
    <p>FARM_EFFECTIVE_MODE_FILE_NAME: <code class="file">.env.production</code></p>
    <p>FARM_BOOL: <code class="bool">true</code></p>
    <p>FARM_EXPAND_A: <code class="expand_a"></code></p>
    <p>CUSTOM_PREFIX_ENV_VARIABLE: <code class="prefix">2</code></p>
    <p>
      FARM_EXPAND_B:
      <code class="expand_b"></code>
    </p>
    <p>
      FARM_ESCAPE_A:
      <code class="escape_a">escape$</code>
    </p>
    <p>FARM_ESCAPE_B: <code class="escape_b">escape$</code></p>
    <p>
      FARM_CUSTOM_ENV_VARIABLE(globalThis process env):
      <code class="globalThis">2</code>
    </p>
    <p>
      FARM_CUSTOM_ENV_VARIABLE(global process env): <code class="global">2</code>
    </p>
    <script>function text(t,e){document.querySelector(t).textContent=e}text(".custom","2"),text(".file",".env.production"),text(".bool","true"),text(".prefix","2"),text(".expand_a",""),text(".expand_b",""),text(".escape_b","escape$"),text(".escape_a","escape$"),text(".globalThis","2"),text(".global","2")</script><script data-farm-entry-script="true">(globalThis||window||global)["4c4d99f12bb13ca32da8c12d962efd60"].__farm_module_system__.setInitialLoadedResources([]),(globalThis||window||global)["4c4d99f12bb13ca32da8c12d962efd60"].__farm_module_system__.setDynamicModuleResourcesMap({}),(globalThis||window||global)["4c4d99f12bb13ca32da8c12d962efd60"].__farm_module_system__.setPublicPaths(["/"]),(globalThis||window||global)["4c4d99f12bb13ca32da8c12d962efd60"].__farm_module_system__.bootstrap()</script>"
`;
